server.go

```diff
@@ -3424,6 +3440,80 @@ func (s *server) findPeerByPubStr(pubStr string) (*peer.Brontide, error) {
 	return peer, nil
 }
 
+func (s *server) sendBackupToPeer(data []byte) error {
+	serverPeers := s.Peers()
+
+	if len(data) > lnwire.MaxPeerStorageBytes {
+		data = data[:lnwire.MaxPeerStorageBytes+1]
+	}
+
+	for _, p := range serverPeers {
+		p.SendBackupMtx.Lock()
+		if p.RemoteFeatures().HasFeature(
+			lnwire.OptionProvideStorageOptional) &&
+			p.SendBackup == 0 {
+
+			if err := p.SendMessage(false, &lnwire.PeerStorage{
+				Blob: data,
+			}); err != nil {
+				return fmt.Errorf("unable to send backup "+
+					"storage to peer(%v), received error: "+
+					"%v", p.PubKey(), err)
+			}
+			p.SendBackup = 1
+			p.LatestPeerBackup = data
+
+		}
+		p.SendBackupMtx.Unlock()
+	}
+
+	return nil
+}
+
+func (s *server) prunePeerBackup(data []byte) error {
+	channels, err := s.chanStateDB.FetchClosedChannels(false)
+	if err != nil {
+		return err
+	}
+
+	var peerBackupToDelete []string
+	for _, channel := range channels {
+		p, err := s.FindPeer(channel.RemotePub)
+		if err != nil {
+			return err
+		}
+
+		bestHeight, err := s.cc.BestBlockTracker.BestHeight()
+
+		if err != nil {
+			return err
+		}
+		if len(s.peerBackup.RetrieveBackupForPeer(p.String())) > 0 &&
+			channel.CloseHeight+minChainBlocksForWipe >=
+				bestHeight {
+
+			peerBackupToDelete = append(peerBackupToDelete, p.
+				String())
+		}
+	}
+
+	return s.peerBackup.PrunePeerStorage(peerBackupToDelete)
+}
```

Comment:
I don't understand what this is supposed to accomplish. This says that if there is a single channel that is past the wipe window for a peer that we delete the storage for that peer?

Reply:
Thanks, maybe adding an extra check of `len(p.activeChannels) == 0` would address the concern?